jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/RetainableByteBuffer.java

```diff
@@ -1411,30 +1439,19 @@ public DynamicCapacity(ByteBufferPool pool, boolean direct, long maxSize, int ag
          */
         public DynamicCapacity(ByteBufferPool pool, boolean direct, long maxSize, int aggregationSize, int minRetainSize)
         {
-            this(new ArrayList<>(), pool, direct, maxSize, aggregationSize, minRetainSize);
+            this(null, new ByteBufferPool.Sized(pool, direct, maxSize > 0 && maxSize < 8192L ? (int)maxSize : aggregationSize), maxSize, minRetainSize);
```

Comment:
Is that `8192L` related to the default in `ByteBufferPool.Sized`? At least this should be made a constant with a meaningful name.